src/main/java/io/grpc/annotations/checkers/AnnotationChecker.java

```diff
@@ -29,13 +29,20 @@
 import com.sun.source.tree.MemberSelectTree;
 import com.sun.source.tree.Tree;
 import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
 import javax.lang.model.element.AnnotationMirror;
+import java.util.Set;
 
 abstract class AnnotationChecker extends BugChecker implements IdentifierTreeMatcher,
     MemberSelectTreeMatcher {
 
   private final String annotationType;
 
+  // When this is set to true, method calls will look for an annotation on the top-level method
+  // declaration. This is used to avoid io.grpc.internal implementations "hiding" publicly declared
+  // API methods.
+  protected boolean checkTopOfMethodHierarchy = false;
```

Comment:
Why not is this variable `final`?

Reply:
I think it could be `private final` because it could be injected via constructor.